### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by crystaljoyce**

### CODE REVIEW

File: db/reviews.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            RETURNING *; 
+        `, [productId]);
+
+        return review; 
+    } catch (error) {
+        throw error; 
+    }
+}
+
+const updateReview = async ({id, title, content, stars, userId, productId}) => {
```
Comment (deannawong) [inline db/reviews.js]: Required - update this database method with fields object instead.

File: db/users.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
 const {client} = require('./client');
 const bcrypt = require('bcrypt');
 
-const createUser = async ({firstName, lastName, email, username, password, address, city, state, zip}) => { 
+const createUser = async ({firstName, lastName, email, username, password, isAdmin=false, address, city, state, zip}) => {
```
Comment (deannawong) [inline db/users.js]: Please make note to update User's database table if default false isn't working.

File: public/index.html [hunk 3]
```diff
@@ -10,4 +10,6 @@
 <body>
   <div id="root"></div>
 </body>
+<footer>
```
Comment (deannawong) [inline public/index.html:13]: There's no content in this footer. Stash it for when there is!

File: routes/index.js [hunk 4]
```diff
@@ -41,6 +45,40 @@ apiRouter.use((req,res,next) => {
   next();
 });
 
+apiRouter.post('/create-checkout-session', async (req, res, next) => {
```
Comment (deannawong) [inline routes/index.js:48]: Please make sure to remove test data when Cart is working.

File: routes/reviews.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        const reviews = await getAllReviews(); 
+
+        res.send(reviews); 
+
+    } catch (error) {
+        next(error);
+    }
+})
+
+reviewsRouter.post('/', requireUser, async (req, res, next) => {
```
Comment (deannawong) [inline routes/reviews.js:17]: Required -  Missing destructuring of contents from req.body.

File: routes/reviews.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    }
+    if(content){
+        updateFields.content = content;
+    }
+    if(stars){
+        updateFields.stars = stars; 
+    }
+
+    try {
+        const reviewToModify = await getReviewById({id})
```
Comment (deannawong) [inline routes/reviews.js]: Required - Make sure the the review is being edited by it's owner.

File: routes/reviews.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            res.send(modifiedReview)
+        } else { 
+            next({message: 'This review could not be modified at this time.'})
+        }
+    } catch (error) {
+        next(error)
+    }
+})
+
+reviewsRouter.delete('/:reviewId', requireUser, async (req, res, next) => {
```
Comment (deannawong) [inline routes/reviews.js:35]: Required - Reviews should only be deleted by it's owner. 
Hint - Create middleware for "requireOwner"

File: src/components/AccountForm.js [hunk 8]
```diff
@@ -177,15 +173,15 @@ const AccountForm = ({type, setToken, setUser}) => {
                         </div>
                         <div>
                             <div>Email</div>
-                            <input type='email' value={email} required onChange={event => setEmail(event.target.value)} ></input>
+                            <input type='text' value={email} required onChange={event => setEmail(event.target.value)} ></input>
```
Comment (deannawong) [inline src/components/AccountForm.js]: Revert  type for input back to email.

File: src/components/Cart.js [hunk 9]
```diff
@@ -1,14 +1,77 @@
...
+import {Link} from 'react-router-dom';
+import axios from 'axios';
+import {SmallProduct} from './Product'
 
-const Cart = ({product}) => {
+const Cart = (props) => {
+    const {token, user} = props
+    const [cart, setCart] = useState({})
+    
+    const fetchCart = async () =>{
```
Comment (deannawong) [inline src/components/Cart.js]: Required -  Keep either fetchCart functions, not both.

**Review by deannawong — CHANGES_REQUESTED** [review]

File: db/users.js (hunk 2 above)
Comment (crystaljoyce) [reply db/users.js]: This has been updated. Thank you!

File: public/index.html (hunk 3 above)
Comment (crystaljoyce) [reply public/index.html:13]: This has been updated. Thank you!

File: routes/reviews.js (hunk 5 above)
Comment (crystaljoyce) [reply routes/reviews.js:17]: This has been updated, thank you!

File: routes/reviews.js (hunk 6 above)
Comment (crystaljoyce) [reply routes/reviews.js]: Yeah, I am planning on doing this on the front end. Edit button will not appear unless the review owner is viewing their own review.

File: routes/reviews.js (hunk 7 above)
Comment (crystaljoyce) [reply routes/reviews.js:35]: Yeah, I am planning on doing this on the front end. Delete button will not appear unless the review owner is viewing their own review.

File: db/reviews.js (hunk 1 above)
Comment (crystaljoyce) [reply db/reviews.js]: This has been updated. Thanks!

**Review by deannawong — APPROVED** [review]
Approved for merging into main (as back-up).
- Missing bug fixes for going to Cart route for exisiting user. 
- Cartpersistence.js file needs to be removed if no longer in use.

**Pull request merged**